python/lsst/summit/extras/plotting/psfPlotting.py

```diff
@@ -0,0 +1,524 @@
+
+    Returns
+    -------
+    cbar : `matplotlib.colorbar.Colorbar`
+        The colorbar object that was added to the axes.
+    """
+    ax = mappable.axes
+    fig = ax.figure
+    divider = make_axes_locatable(ax)
+    cax = divider.append_axes("right", size="5%", pad=0.05)
+    cbar = fig.colorbar(mappable, cax=cax)
+    return cbar
+
+
+def makeTableFromSourceCatalogs(icSrcs: dict[int, SourceCatalog], visitInfo: VisitInfo) -> Table:
+    """Extract the shapes from the source catalogs into an astropy table.
+
+    The shapes of the PSF candidates are extracted from the source catalogs and
+    transformed into the required coordinate systems for plotting either focal
+    plane coordinates, az/el coordinates, or equatorial coordinates.
+
+    Parameters
+    ----------
+    icSrcs : `dict` [`int`, `lsst.afw.table.SourceCatalog`]
+        A dictionary of source catalogs, keyed by the detector numbers.
+    visitInfo : `lsst.afw.image.VisitInfo`
+        The visit information for a representative visit.
+
+    Returns
+    -------
+    table : `astropy.table.Table`
+        The table containing the data from the source catalogs.
+    """
+    tables = []
+
+    for detectorNum, icSrc in icSrcs.items():
+        icSrc = icSrc.asAstropy()
+        icSrc = icSrc[icSrc["calib_psf_candidate"]]
+        icSrc["detector"] = detectorNum
+        tables.append(icSrc)
+
+    table = vstack(tables)
+    # Add shape columns
+    table["Ixx"] = table["slot_Shape_xx"] * (0.2) ** 2
+    table["Ixy"] = table["slot_Shape_xy"] * (0.2) ** 2
+    table["Iyy"] = table["slot_Shape_yy"] * (0.2) ** 2
+    table["T"] = table["Ixx"] + table["Iyy"]
+    table["e1"] = (table["Ixx"] - table["Iyy"]) / table["T"]
+    table["e2"] = 2 * table["Ixy"] / table["T"]
+    table["e"] = np.hypot(table["e1"], table["e2"])
+    table["x"] = table["base_FPPosition_x"]
+    table["y"] = table["base_FPPosition_y"]
+
+    table.meta["rotTelPos"] = (
+        visitInfo.boresightParAngle - visitInfo.boresightRotAngle - (np.pi / 2 * radians)
+    ).asRadians()
+    table.meta["rotSkyPos"] = visitInfo.boresightRotAngle.asRadians()
+
+    rtp = table.meta["rotTelPos"]
+    srtp, crtp = np.sin(rtp), np.cos(rtp)
+    aaRot = np.array([[crtp, srtp], [-srtp, crtp]]) @ np.array([[0, 1], [1, 0]]) @ np.array([[-1, 0], [0, 1]])
+    table = extendTable(table, aaRot, "aa")
+    table.meta["aaRot"] = aaRot
+
+    rsp = table.meta["rotSkyPos"]
+    srsp, crsp = np.sin(rsp), np.cos(rsp)
+    nwRot = np.array([[crsp, -srsp], [srsp, crsp]])
+    table = extendTable(table, nwRot, "nw")
+    table.meta["nwRot"] = nwRot
+
+    return table
+
+
+def extendTable(table: Table, rot: npt.NDArray[np.float_], prefix: str) -> Table:
+    """Extend the given table with additional columns for the rotated shapes.
+
+    Parameters
+    ----------
+    table : `astropy.table.Table`
+        The input table containing the original shapes.
+    rot : `np.ndarray`
```

Comment:
```suggestion
    rot : `numpy.ndarray`
```
Sphinx's linking doesn't pay attention to how you've imported a module.